### PR TITLE
Chore: port `ConsumerRemovalProposal` handler to main

### PR DESCRIPTION
**Pull request by sainoe**
Port `ConsumerRemovalProposal` proposal handler and cli command from [v0.1.4](https://github.com/cosmos/interchain-security/releases/tag/v0.1.4) to main

### CODE REVIEW

**Comment by smarshall-spitzbart** [comment]
Nice catch, thanks!

**Comment by MSalopek** [comment]
Added integration tests.

File: tests/integration/actions.go [hunk 1]
```diff
@@ -247,6 +248,65 @@
...
+	}
+
+	jsonStr := string(bz)
+	if strings.Contains(jsonStr, "'") {
+		log.Fatal("prop json contains single quote")
+	}
+
+	//#nosec G204 -- Bypass linter warning for spawning subprocess with cmd arguments.
+	bz, err = exec.Command("docker", "exec", tr.containerConfig.instanceName,
+		"/bin/bash", "-c", fmt.Sprintf(`echo '%s' > %s`, jsonStr, "/temp-proposal.json")).CombinedOutput()
```
[the remark this answers is not in the document]
Comment (smarshall-spitzbart) [reply tests/integration/actions.go:284]: closed this alert, seen it before. Shouldn't matter for tests

File: tests/integration/state.go [hunk 2]
```diff
@@ -36,14 +36,25 @@ type TextProposal struct {
...
-type ConsumerProposal struct {
+type ConsumerAdditionProposal struct {
 	Deposit       uint
 	Chain         chainID
 	SpawnTime     int
 	InitialHeight clienttypes.Height
 	Status        string
 }
 
+func (p ConsumerAdditionProposal) isProposal() {}
```
[the remark this answers is not in the document]
Comment (MSalopek) [reply tests/integration/state.go:47]: Yes. Proposal interface has to be implicitly implemented.
https://github.com/cosmos/interchain-security/blob/main/tests/integration/state.go#L27

File: tests/integration/actions.go [hunk 3]
```diff
@@ -994,3 +1054,46 @@ func (tr TestRun) invokeDoublesignSlash(
 	}
 	tr.waitBlocks("provi", 10, 2*time.Minute)
 }
+
+type queryConsumerChainRemovedAction struct {
```
[the remark this answers is not in the document]
Comment (MSalopek) [reply tests/integration/actions.go]: refactored and removed action

File: tests/integration/actions.go [hunk 4]
```diff
@@ -247,6 +248,65 @@ func (tr TestRun) submitConsumerAdditionProposal(
 	}
 }
 
+type submitConsumerRemovalProposalAction struct {
+	chain         chainID
+	from          validatorID
+	deposit       uint
+	consumerChain chainID
+	stopTime      uint
```
[the remark this answers is not in the document]
Comment (MSalopek) [reply tests/integration/actions.go]: Refactored

File: tests/difference/core/node_modules/.yarn-integrity [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{
```
[the remark this answers is not in the document]
Comment (MSalopek) [reply tests/difference/core/node_modules/.yarn-integrity]: removed

File: tests/integration/state.go [hunk 6]
```diff
@@ -473,9 +473,8 @@ func (tr TestRun) getParam(chain chainID, param Param) string {
 	return value.String()
 }
 
-// queryConsumerChainRemoved checks if provided consumerChain
-// was removed from the list of consumer chains on the provider.
-// If a chain was not removed a panic will be raised.
+// getConsumerChains returns a list of consumer chains that're being secured by the provider chain,
```
Comment (MSalopek) [inline tests/integration/state.go:476]: Thanks!

**Review by smarshall-spitzbart — APPROVED** [review]
couple small nits, added them myself, LGTM 🚀

File: tests/integration/steps_start_chains.go [hunk 7]
```diff
@@ -82,7 +82,7 @@ func stepsStartChains(consumerName string, setupTransferChan bool) []Step {
 			state: State{
 				chainID("provi"): ChainState{
 					Proposals: &map[uint]Proposal{
-						1: ConsumerProposal{
+						1: ConsumerAdditionProposal{
```
Comment (jtremback) [inline tests/integration/steps_start_chains.go:85]: @MSalopek great opportunity to improve these consumer chain addition tests by adding in your new `ConsumerChains: &map[chainID]bool{"consu": true}` state accessor!

**Review by jtremback — APPROVED** [review]
LGTM

**Pull request merged**